docker/docker/src/main/java/org/arquillian/cube/docker/impl/await/HealthCheckBeforeClassObserver.java

```diff
@@ -0,0 +1,118 @@
+package org.arquillian.cube.docker.impl.await;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+import org.arquillian.cube.HealthCheck;
+import org.arquillian.cube.docker.impl.docker.DockerClientExecutor;
+import org.arquillian.cube.docker.impl.util.Ping;
+import org.arquillian.cube.impl.util.ReflectionUtil;
+import org.arquillian.cube.impl.util.Timespan;
+import org.arquillian.cube.spi.Cube;
+import org.arquillian.cube.spi.CubeRegistry;
+import org.arquillian.cube.spi.metadata.HasPortBindings;
+import org.jboss.arquillian.core.api.Instance;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.core.api.annotation.Observes;
+import org.jboss.arquillian.test.spi.TestClass;
+import org.jboss.arquillian.test.spi.event.suite.BeforeClass;
+
+public class HealthCheckBeforeClassObserver {
+
+    @Inject
+    Instance<DockerClientExecutor> dockerClientExecutorInstance;
+
+    @Inject
+    Instance<CubeRegistry> cubeRegistryInstance;
+
+    public void executeHealthCheck(@Observes BeforeClass beforeClass) {
+        final TestClass testClass = beforeClass.getTestClass();
+        if (ReflectionUtil.isClassWithAnnotation(testClass.getJavaClass(), HealthCheck.class)) {
+
+            final HealthCheck healthCheck = testClass.getAnnotation(HealthCheck.class);
+            executeHealthCheck(healthCheck);
+
+        }
+    }
+
+    private boolean executeHealthCheck(HealthCheck healthCheck) {
+
+        final URL url = buildUrl(healthCheck);
+        final int pollIterations = healthCheck.iterations();
+        final long sleepTime = Timespan.toMilliseconds(healthCheck.interval());
+        final int responseCode = healthCheck.responseCode();
+        final String method = healthCheck.method();
+        final Long connectTimeout = Timespan.toMilliseconds(healthCheck.timeout());
+
+        return Ping.ping(pollIterations, sleepTime, TimeUnit.MILLISECONDS, () -> {
+            try {
+
+                HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
```

Comment:
Shouldn't we close this connection afterwards? 
Something to investigate maybe https://scotte.org/2015/01/httpurlconnection-socket-leak/